src/parser/parser.cc

```diff
@@ -1130,16 +1146,25 @@ class Parser {
         ret_type = ParseType();
       }
 
+      ObjectRef virtual_device;
+      if (WhenMatch(TokenType::kComma)) {
+        virtual_device = ParseVirtualDevice();
```

Comment:
Ideal would be if we could parse the attributes as usual, then 'promote' the "virtual_device" attribute to the first class field. Since meta references are handled by ParseAttributeValue I think that would work out.

Reply:
OK, can do that